app/models/middleware_performance.rb

```diff
@@ -0,0 +1,74 @@
+class MiddlewarePerformance < ActsAsArModel
+  set_columns_hash(
+    :id            => :string,
+    :name          => :string,
+    :resource_type => :string,
+    :feed          => :string,
+    :ems_ref       => :string,
+    :start_time    => :datetime,
+    :end_time      => :datetime
+  )
+
+  alias resource_id id
+  alias resource_name name
+
+  def self.build_results_for_report_middleware(options)
+    start_time, end_time, interval = parse_time_interval(options)
+    results = []
+    model_class.find_each do |ms|
```

Comment:
> As this abstract class does not define the model_class (but the subclasses do), can we add a check here that it is indeed set and otherwise return a meaningful error message?
> I think that customers can create their own report definitions and they may then also try to use the MiddlewarePerformance in the db:xx expression

good point. I do know that customers can create their own custom reports typically by adding some yaml files (saw couple of db schemas where there were much more reports than the default)